dhcpv4/ztpv4/ztp.go

```diff
@@ -76,5 +74,5 @@ func ParseVendorData(packet *dhcpv4.DHCPv4) (*VendorData, error) {
 	}
 
 	// We didn't match anything.
-	return nil, nil
+	return nil, errors.New("failed to parse vendor option data")
```

Comment:
nit: this is not a parsing failure, rather no known ZTP vendor is found, I would change the error string